### ISSUE TITLE
[Bug]: Perpetual scan on engine detach

### ISSUE THREAD

**Issue by Ibetz1** [boskokg/flutter_blue_plus#657]
### Requirements

- [X] I've looked at the README 'Common Problems' section

### Have you checked this problem on the example app?

Yes

### FlutterBluePlus Version

1.20.7

### Flutter Version

3.13.6

### What OS?

Android

### OS Version

Android 13

### Bluetooth Module

esp32-s1

### What is your problem?

On abruptly closing the app during a scan, it never stops the scan causing this warning "W/[FBP-Android](10659): [FBP] invokeMethodUIThread: tried to call method on closed channel: OnScanResponse".

Upon reopening the app, the thread throwing the warning appears to be stuck open. I fixed the error myself in your "FlutterBluePlusPlugin.java" in the "onDetachedFromEngine" method by adding:

```java
// ON APP CLOSE
@Override
public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding)
{
    log(LogLevel.DEBUG, "onDetachedFromEngine");

    pluginBinding = null;

    disconnectAllDevices("onDetachedFromEngine");

    // perpetual scan fix //
    BluetoothLeScanner scanner = mBluetoothAdapter.getBluetoothLeScanner();
    scanner.stopScan(getScanCallback());
    // end perpetual scan fix //
    
    context.unregisterReceiver(mBluetoothBondStateReceiver);
    context.unregisterReceiver(mBluetoothAdapterStateReceiver);
    context = null;

    methodChannel.setMethodCallHandler(null);
    methodChannel = null;

    mBluetoothAdapter = null;
    mBluetoothManager = null;
}
```

I am not sure if this does this for IOS but it might be something to look into, I also don't know if this is a proper fix or not.

### Logs

```shell
W/[FBP-Android](10659): [FBP] invokeMethodUIThread: tried to call method on closed channel: OnScanResponse (this gets looped infinitely)
```

**Comment by chipweinberger** [comment]
seems like a reasonable fix

**Comment by chipweinberger** [comment]
fixed 1.24.1

**Comment by Ibetz1** [comment]
sweet thanks
